### PR TITLE
Basic Mob Flesh Worm

### LINKED ISSUE #68614: Flesh worm segments disappear after entering a transit tube
RoundID 186990

be funny worm with many segments
enter transit tube, they suddenly disappear and will not come back even after de worming and worming again

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Fixes #68614

Converts the Flesh Worm (Armsy) into a Basic Mob.
Most of its behaviour has been moved into a component which we can use to make arbitrary mobs into linked lists of mobs.
To accomplish this I added a signal which is sent when you call any `adjustXLoss` proc, let me know if my implementation is "calling the same signal from several places" by a backdoor, I wanted to avoid registering to 6 signals but I'll change it if I must.

While I was here I killed the unused "lesser" variant because we stopped using it. Resultingly, Ascended Armsy doesn't need to distinguish itself by inflating the sprite, so it doesn't. This means that now flesh worms are using their sprites as intended to be displayed, but if people really miss all of its segments being poorly scaled by the byond engine then I guess I can restore it.

## Why It's Good For The Game

dreamseeker_p8vOpZGXII

## Changelog

:cl:
refactor: Flesh Worms are now basic mobs. Please report any unexpected behaviour.
sprite: Flesh Worms are a little bit slimmer.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
closes #68614?

**Comment by Jacquerel** [comment]
> closes #68614?

dreamseeker_KTk7o7dpYG
Looks a bit weird but seems fixed.

File: code/modules/admin/verbs/debug.dm [hunk 1]
```diff
@@ -948,6 +948,51 @@
...
+	if (!check_rights(R_DEBUG))
+		return
+	if(isnull(holder))
+		return
+	if(!isliving(holder.marked_datum))
+		to_chat(usr, span_warning("Error: Please mark a mob to attach mobs to."))
+		return
+	var/mob/living/head = holder.marked_datum
+
+	var/attempted_target_path = input(
```
Comment (san7890) [inline code/modules/admin/verbs/debug.dm:964]: making worms out of people isn't an admin-critical tool (sadly), make this `tgui_input` instead. you use `tgui_input_number` later on so lets be consistent

File: code/modules/admin/verbs/debug.dm [hunk 2]
```diff
@@ -948,6 +948,51 @@
 /proc/cmp_timer_data(list/a, list/b)
 	return b["count"] - a["count"]
 
+/client/proc/create_mob_worm()
+	set category = "Debug"
+	set name = "Create Mob Worm"
+	set desc = "Attached a linked list of mobs to a marked mob"
+	if (!check_rights(R_DEBUG))
```
Comment (san7890) [inline code/modules/admin/verbs/debug.dm:955]: `R_FUN` instead imo (or both) (people with `R_DEBUG` can give themselves fun but shhh)

File: code/modules/antagonists/heretic/structures/knock_final.dm [hunk 3]
```diff
@@ -21,8 +21,7 @@
 	/// A static list of heretic summons which we should not create
 	var/static/list/monster_types_blacklist = list(
 		/mob/living/basic/heretic_summon/star_gazer,
-		/mob/living/simple_animal/hostile/heretic_summon/armsy,
-		/mob/living/simple_animal/hostile/heretic_summon/armsy/prime,
+		/mob/living/basic/heretic_summon/armsy,
```
Comment (san7890) [inline code/modules/antagonists/heretic/structures/knock_final.dm:24]: alphabetize

File: code/datums/components/mob_chain.dm [hunk 4]
```diff
@@ -0,0 +1,205 @@
+/**
+ * Component allowing you to create a linked list of mobs.
+ * These mobs will follow each other and attack as one, as well as sharing damage taken.
+ */
+/datum/component/mob_chain
```
Comment (san7890) [inline code/datums/components/mob_chain.dm:5]: human centipede simulator just in time for halloween (no change)

**Review by san7890 — CHANGES_REQUESTED** [review]
nits iirc

**Comment by LemonInTheDark** [comment]
> I added a signal which is sent when you call any `adjustXLoss` proc, let me know if my implementation is "calling the same signal from several places" by a backdoor, I wanted to avoid registering to 6 signals but I'll change it if I must.

I think that'd be a better pattern TBH. You aren't the only one who might want to access this info, and they might want it split up by type. Plus you could just feed them all into one proc, so it's just a RegisterSignals and not RegisterSignal

**Comment by Jacquerel** [comment]
I'll go back and do it

**Comment by LemonInTheDark** [comment]
I appreciate it. don't have to if you don't wanna. I won't hold it against you I just saw this on the bus and was like "wait this is bad", and wanted to say somethin when I got to computer.

**Pull request merged**